src/DataTransferObject.php

```diff
@@ -318,6 +321,40 @@ public function getDefinedPropertyNames(): array
         return array_keys($this->properties);
     }
 
+    /**
+     * @param string|array $propertyNames
+     *
+     * @return void
+     */
+    public function assertDefined($propertyNames): void
+    {
+        $this->assertKnownPropertyNames($propertyNames);
+
+        $undefined = array_filter((array) $propertyNames, function (string $propertyName) {
+            return !$this->isDefined($propertyName);
+        });
```

Comment:
If the isDefined check is no longer as complicated as it used to be (i imagine it's not now that nothing is defined unless its actually initialised) - again might be more performant to just do a array_diff.

`$undefined = array_diff($propertyNames, array_keys($this->properties));`

Reply:
Unfortunately because is defined supports dot notation we still have to call it.